web-common/src/Component/Box.purs

```diff
@@ -0,0 +1,44 @@
+module Component.Box
+  ( Preset(..)
+  , box
+  , boxIn
+  ) where
+
+import Component.Container
+  ( Container
+  , ContainerOf
+  , container
+  )
+import Data.Default (class Default)
+import Halogen.HTML as HH
+
+data Preset
+  = NoSpace
+  | Cramped
+  | Snug
+  | Default
+  | Loose
+  | CardCramped
+  | CardSnug
+  | Card
+  | CardLoose
+
+instance defaultPreset :: Default Preset where
+  default = Default
+
+boxIn :: Preset -> ContainerOf
+boxIn preset = container baseClasses
+  where
+  baseClasses = case preset of
+    NoSpace -> [ "p-0" ]
```

Comment:
not for now, but I think I prefer having the CSS directly inline instead of the Presets, as it implies another level of indirection

Reply:
I will refactor this after merging